core/frontend/src/common/imdl/ParseImdlDocument.ts

```diff
@@ -299,11 +304,29 @@ class Parser {
     this._timeline = options.timeline;
   }
 
-  public parse(): Imdl.Document | ImdlParseError {
+  public async parse(): Promise<Imdl.Document | ImdlParseError> {
     const featureTable = this.parseFeatureTable();
     if (!featureTable)
       return TileReadStatus.InvalidFeatureTable;
 
+    if (this.hasMeshoptCompression()) {
+      if (this._options.meshoptCompressionNotSupported || !MeshoptDecoder.supported)
+        return TileReadStatus.UnsupportedMeshoptCompression;
+
+      if (meshoptDecoderStatus === "uninitialized") {
```

Comment:
@pmconne I ended up doing things a bit like in MeshoptCompression.ts except that the module is loaded statically instead, do you reckon that we should perhaps make MeshCompression.ts use static import and use it here instead ?